sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Utils.java

```diff
@@ -34,7 +35,13 @@
     private static final JsonParser sParser = new JsonParser();
 
     static <T> Document<T> parseDocument(String cosmosDbPayload, Class<T> documentType) {
-        return parseDocument(sParser.parse(cosmosDbPayload).getAsJsonObject(), documentType);
+        JsonObject body;
+        try {
+            body = sParser.parse(cosmosDbPayload).getAsJsonObject();
+        } catch (Exception e) {
```

Comment:
RuntimeException, it's not checked so we should not catch knives.

Reply:
Understood, change made.